### ISSUE TITLE
jsx-no-bind: consider local function declaration usage a violation

### ISSUE THREAD

**Issue by p7g** [yannickcr/eslint-plugin-react#3047]
Currently the `jsx-no-bind` rule allows the following:

```jsx
function MyComponent() {
  function doStuff() {}
  return <OtherComponent onClick={doStuff} />;
}
```

I think it would be beneficial if it would consider that to be a violation, since I see this done relatively often.

Was it a decision to ignore function declarations (and `let` and `var`) to avoid needing extra analyses to do with reassignment? If not, would a pull request to treat `function a() {}` the same as `const a = function () {}` be accepted?

**Comment by ljharb** [comment]
It wasn't intentional. A PR for this would definitely be accepted.
